modules/ting_das/includes/ting_das.client.inc

```diff
@@ -0,0 +1,430 @@
+    return $this->userName;
+  }
+
+  /**
+   * Set the users e-mail-address.
+   *
+   * @param string $mail
+   *   The users e-mail-address.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserMail(string $mail) {
+    $this->mail = $mail;
+    return $this;
+  }
+
+  /**
+   * Get the users e-mail-address.
+   *
+   * @return string|null
+   *   The users e-mail-address.
+   */
+  public function getUserMail() {
+    return $this->mail;
+  }
+
+  /**
+   * The interest date for the user.
+   *
+   * The date until the user is not longer interested in getting the material.
+   *
+   * @param string $userInterestDays
+   *   The users interest date in number of days.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserInterestDate(string $userInterestDays) {
+    $this->userInterestDays = $userInterestDays;
+    return $this;
+  }
+
+  /**
+   * Get the users interest date.
+   *
+   * @param string $format
+   *   The date format to use.
+   *
+   * @return string|null
+   *   The users interest date in the format given from now.
+   */
+  public function getUserInterestDate($format = 'Y-m-d') {
+    $date = strtotime('+' . $this->userInterestDays . ' day');
+    return date($format, $date);
+  }
+
+  /**
+   * Set the agency id for pickup.
+   *
+   * @param string $pickupAgencyId
+   *   The pickup agency id.
+   *
+   * @Note: Not sure why this exists as all articles are sent by mail to the
+   *        end user.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPickupAgencyId(string $pickupAgencyId) {
+    $this->pickupAgencyId = $pickupAgencyId;
+    return $this;
+  }
+
+  /**
+   * Get pickup agency id.
+   *
+   * @return string|null
+   *   The pickup agency id.
+   */
+  public function getPickupAgencyId() {
+    return $this->pickupAgencyId;
+  }
+
+  /**
+   * Set agency id.
+   *
+   * @param string $agencyId
+   *   The agency id.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setAgencyId(string $agencyId) {
+    $this->agencyId = $agencyId;
+    return $this;
+  }
+
+  /**
+   * Get agency id.
+   *
+   * @return string|null
+   *   The agency id.
+   */
+  public function getAgencyId() {
+    return $this->agencyId;
+  }
+
+  /**
+   * The title of the publication.
+   *
+   * @param string $publicationTitle
+   *   The publications title.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPublicationTitle(string $publicationTitle) {
+    $this->publicationTitle = $publicationTitle;
+    return $this;
+  }
+
+  /**
+   * Get article title.
+   *
+   * @return string|null
+   *   The article title.
+   */
+  public function getPublicationTitle() {
+    return $this->publicationTitle;
+  }
+
+  /**
+   * Set the date of publication.
```

Comment:
I think this warrants some explanation regarding what the date of the publication means. Especially since we also have a publication year.

We could let clients pass a `DateTime` object and just extract date + year internally if we need to send it as separate values.

Reply:
The service is un-documented so your guess is as good as mine.

It's here as it's possible to set it.